src/main/java/life/qbic/subscriptions/security/SecurityConfiguration.java

```diff
@@ -0,0 +1,52 @@
+package life.qbic.subscriptions.security;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.http.HttpMethod;
+import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.builders.WebSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.config.http.SessionCreationPolicy;
+
+@Configuration
+@EnableWebSecurity
+public class SecurityConfiguration extends WebSecurityConfigurerAdapter {
+
+  private static final String REALM="QBIC";
+
+  @Autowired
+  public void configureGlobalSecurity(
+      AuthenticationManagerBuilder auth,
+      @Value("${service.officer.name}") String officerName,
+      @Value("${service.officer.password}") String officerPassword) throws Exception {
+    auth.inMemoryAuthentication()
+        .withUser(officerName).password("{noop}"+officerPassword).roles("OFFICER");
+
+  }
+
+  @Override
+  protected void configure(HttpSecurity http) throws Exception {
+    http.csrf().disable()
+        .authorizeRequests()
+        .antMatchers(HttpMethod.GET, "/subscription/cancel").hasRole("OFFICER")
+        .and().httpBasic().realmName(REALM).authenticationEntryPoint(getBasicAuthEntryPoint())
+        .and().sessionManagement().sessionCreationPolicy(SessionCreationPolicy.STATELESS);//We don't need sessions to be created.
```

Comment:
here we secure the endpoint and remove session management ( we dont need it for our use case). Since we have no web form, we need to configure Spring in such a way, that the it knows what to call when the authentication fails.

If you go into the `QbicBasicAuthenticationEntryPoint.class` you see that we handle there, which will result in a Http 401 response.